pages/preimages/[id].tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Pagination, Skeleton } from 'antd';
+import { GetServerSideProps } from 'next';
+import dynamic from 'next/dynamic';
+import { useRouter } from 'next/router';
+import { getPreimages, IPreimagesListingResponse } from 'pages/api/v1/listing/preimages';
+import React, { FC, useEffect } from 'react';
+
+import { getNetworkFromReqHeaders } from '~src/api-utils';
+import { useNetworkContext } from '~src/context';
+import { LISTING_LIMIT } from '~src/global/listingLimit';
+import SEOHead from '~src/global/SEOHead';
+import { ErrorState } from '~src/ui-components/UIStates';
+import { handlePaginationChange } from '~src/util/handlePaginationChange';
+
+const PreImagesTable = dynamic(() => import('~src/components/PreImagesTable'), {
+	loading: () => <Skeleton active /> ,
+	ssr: false
+});
+
+export const getServerSideProps: GetServerSideProps = async ({ req, query }) => {
+	const { page = 1 } = query;
+	const network = getNetworkFromReqHeaders(req.headers);
+
+	const { data, error } = await getPreimages({
```

Comment:
getPreimages should accept a single hash and return data just for that hash for this file